[docs] Add Workflows and Maestro insights docs (#46447)
expo/expo: docs/pages/eas-insights/app-usage.mdx

---
title: App usage

> sidebar_title: App usage

Reviewer: ```suggestion
```

Sidebar title is optional and useful when the title of the page is lengthy. In this case, both are same so the value of `title` is used in the sidebar.

description: View your app's usage across platforms, app store versions, and time in the App usage tab of EAS Insights.
---

import { ContentSpotlight } from '~/ui/components/ContentSpotlight';
import { Terminal } from '~/ui/components/Snippet';

> **important** **EAS Insights** is in [preview](/more/release-statuses/#preview) and subject to breaking changes. While in preview, it is free to use.

> The **App usage** tab offers a view into a project's performance, usage, and reach. It makes it easy to see the state of your app, providing information about usage across platforms, app store versions, and timeframes.

Reviewer: ```suggestion
The **App usage** tab offers a view into a project's performance, usage, and reach. It allows you to see the state of your app, providing information about usage across platforms, app store versions, and timeframes.
```

<ContentSpotlight
  alt="The App usage tab in EAS Insights showing app usage across platforms, app store versions, and time."
  src="/static/images/eas-insights/insights-app-usage-light.png"
  darkSrc="/static/images/eas-insights/insights-app-usage-dark.png"
/>

## Integration with EAS Update

> If you're already using [EAS Update](/eas-update/introduction/), we provide certain high-level usage insights without any additional client-side changes. This is possible because by aggregating data from client requests to check for an update into a limited Insights view that shows usage over time, as well as usage broken down by platform.

Reviewer: **[suggestion]** Non-tutorial pages avoid first-person plural (`expo-docs-style-guide.md#voice-and-tone`). This sentence moved verbatim from `introduction.mdx`, where this PR dropped the first-person plural, so aligning the relocated copy keeps the page consistent with that edit.

```suggestion
If you're already using [EAS Update](/eas-update/introduction/), you get certain high-level usage insights without any additional client-side changes. This is possible because by aggregating data from client requests to check for an update into a limited Insights view that shows usage over time, as well as usage broken down by platform.
```

> If you're already using [EAS Update](/eas-update/introduction/), we provide certain high-level usage insights without any additional client-side changes. This is possible because by aggregating data from client requests to check for an update into a limited Insights view that shows usage over time, as well as usage broken down by platform.

Reviewer: **[suggestion]** A second note on this line, complementing the first-person-plural suggestion already here. The sentence "This is possible because by aggregating data..." has no main clause, so it reads as a fragment (`expo-docs-style-guide.md#core-principles`). When you apply the other suggestion on this line, tighten this too. Possible rewrite: This works by aggregating the update-check requests your app already makes into a limited Insights view of usage over time and by platform.

## Use the `expo-insights` library

> Developers can add the `expo-insights` library to their projects and gain more precise usage metrics (than those provided by just aggregating update requests) and additional breakdowns by app store version. Currently, the library is limited to sending client events only pertaining to cold starts of the app, but in the future we will expand `expo-insights` to offer new types of events and payloads, to support more advanced functionality.

Reviewer: **[suggestion]** Same first-person-plural cleanup (`expo-docs-style-guide.md#voice-and-tone`). Also relocated verbatim from the old introduction.

```suggestion
Developers can add the `expo-insights` library to their projects and gain more precise usage metrics (than those provided by just aggregating update requests) and additional breakdowns by app store version. Currently, the library is limited to sending client events only pertaining to cold starts of the app, but in the future, `expo-insights` will expand to offer new types of events and payloads, to support more advanced functionality.
```

### Installation

> To use the `expo-insights`, make sure your app is linked to your EAS project in **app.json** / **app.config.js** by running `eas init`, then install the library.

Reviewer: **[suggestion]** Reads as if a word is missing after "the" here (`expo-docs-style-guide.md#core-principles`). Drop the article or add a noun. Relocated from the old introduction.

```suggestion
To use `expo-insights`, make sure your app is linked to your EAS project in [app config](/workflow/configuration/) by running `eas init`, then install the library.
```

Currently, there are 4 variants of app config file which makes the above reference incorrect: https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#referencing-appjsonappconfigjsonappconfigjsappconfigts

<Terminal
  cmd={[
    '# Install EAS CLI if you have not already',
    '$ npm i -g eas-cli',
    '',
    '# Initialize your project EAS if you have not already ',
    '$ eas init',
    '',
    '# Install the library',
    '$ npx expo install expo-insights',
  ]}
/>

> After installing the library, create a build either with [EAS](/build/setup/) or [locally](/guides/local-app-development/). The library will automatically send events to EAS Insights when the app is launched.

Reviewer: **[suggestion]** Use present tense and active voice (`expo-docs-style-guide.md#voice-and-tone`): "automatically sends" and "when the app launches" instead of the future passive. Relocated from the old introduction.

```suggestion
After installing the library, create a build either with [EAS](/build/setup/) or [locally](/guides/local-app-development/). The library automatically sends events to EAS Insights when the app launches.
```
